### ISSUE TITLE
Relation schema does not respect direction / no error on wrong node labels / `CREATE` rewrites user-provided node label

### ISSUE THREAD

**Issue by sapalli2989** [kuzudb/kuzu#2675]
Version: v0.1.0

Schema:
```cypher
CREATE NODE TABLE V1 (id STRING, PRIMARY KEY(id));
CREATE NODE TABLE V2 (id STRING, PRIMARY KEY (id));
CREATE REL TABLE E(FROM V1 TO V2);
```
Case 1:
```cypher
kuzu> CREATE (v21: V2 {id: "v21"})-[:E]->(v11:V1 {id: "v11"});
-
(0 tuples)

kuzu> MATCH (v:V2) RETURN v;
-----------------------------------
| v                               |
-----------------------------------
| {_ID: 1:0, _LABEL: V2, id: v21} |
-----------------------------------
(1 tuple)

kuzu> MATCH (v:V1) RETURN v;
-----------------------------------
| v                               |
-----------------------------------
| {_ID: 0:0, _LABEL: V1, id: v11} |
-----------------------------------
(1 tuple)

kuzu> MATCH (v2:V2)-[e:E]->(v1:V1) RETURN v2,e,v1;
Error: Binder exception: Cannot find a label for node v2 that connects to all of its neighbour relationships.

kuzu> MATCH (v1:V1)-[e:E]->(v2:V2) RETURN v1,e,v2;
----------------------------------------------------------------------------------------------------------
| v1                              | e                                  | v2                              |
----------------------------------------------------------------------------------------------------------
| {_ID: 0:0, _LABEL: V1, id: v11} | (0:0)-{_LABEL: E, _ID: 2:0}->(1:0) | {_ID: 1:0, _LABEL: V2, id: v21} |
----------------------------------------------------------------------------------------------------------
(1 tuple)
```

Expected: `CREATE` errors, as the direction `V2 -> V1` is wrong, schema is `FROM V1 TO V2`.
Actual: Both nodes are created with the inverse direction. Querying the actual user-provided direction results in not-understandable `Error: Binder exception`.

Case 2:
```cypher
kuzu> CREATE (v22:V2 {id: "v22"})-[:E]->(v23: V2 {id: "v23"});
-
(0 tuples)

kuzu> MATCH (v2a:V2)-[e:E]->(v2b:V2) RETURN v2a,e,v2b;
Error: Binder exception: Cannot find a label for node v2a that connects to all of its neighbour relationships.
```
Same as case 1, no error from REL schema. But in this case, inproper inverse relation is not created.

**Comment by andyfengHKU** [comment]
Hi @sapalli2989, a fix is now available in #2680.

I also improve the error message to report what are the expected label candidates that match schema constraint.
```
-STATEMENT CREATE (v21: V2 {id: "v21"})-[:E]->(v11:V1 {id: "v11"});
 ---- error
 Binder exception: Query node v21 violates schema. Expected labels are V1.
```

**Comment by sapalli2989** [comment]
@andyfengHKU Appreciate the quick fix, thanks!
